src/components/Form/Form.jsx

```diff
@@ -15,12 +16,37 @@ export const Form = () => {
     lastDate: null,
   });
 
+  const [value] = useCollection(getAll());
+
+  const isItemDuplicated = (name) => {
+    const regex = /[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~]/g;
+
+    if (value) {
+      const item = value.docs.filter(
+        (doc) =>
+          doc.data().name.toLowerCase().replace(regex, '') ===
+          name.toLowerCase().replace(regex, ''),
+      );
+      return item.length ? true : false;
+    } else {
+      return false;
+    }
+  };
+
   const sendToFB = (e) => {
     e.preventDefault();
-    const itemLength = values?.nameItem;
+    const itemName = values?.nameItem;
     const selectLength = values?.selectTime;
 
-    if (itemLength && selectLength) {
+    if (isItemDuplicated(itemName)) {
+      setError('Item is already on the list');
+      setTimeout(() => {
```

Comment:
This is another example of that memory leak we mentioned last Sunday. Cleaning these up sounds complicated, but it's actually not too bad.

`setTimeout` returns an id, which we can use to cancel it with `clearTimeout`. https://www.w3schools.com/jsref/met_win_cleartimeout.asp

Where do we cancel it? 
`useEffect` if we return a function in the `useEffect` hook, it will be run when the component unmounts. This is so that we can clean things like this up in our components and not use up memory unnecessarily. 

How in the world do we save the id to be able to call it later in the return value of `useEffect`?   
You may be tempted to save the id as state and that would totally work, but I would actually recommend saving it as a "ref". If you don't know what they are, ref's are references to a specific memory address. They are fully mutable and updating them will not cause a re-render. If we were to save the timeout as state, then we would be causing a re-render. We shouldn't overuse refs, because they can get messy, but for saving something like a timeout id, I think it makes sense. To create the ref, we use the `useRef` hook and then assign the timeout id to the `current` property on the ref.

So, what does all this look like.

```javascript
const timeoutIdRef = useRef();

useEffect(() => {
  return () => { 
     if(timoutIdRef.current) clearTimout(timeoutIdRef.current);
  }
}, [])

// I'm assuming "FB stands for Facebook by the way". Abbreviations can cause confusion. ;-)
const sendToFacebook = (e) => {
   ...
   timeoutIdRef.current = setTimeout(() => { ...
}
```
If you've got the time, you can try fixing the other memory leaks as well? or...you could try your hand at making something like a `useError` custom-hook as I see we're using the self-cleaning error a few spots in the app. Just a thought.